dags/shredder_backfill.py

```diff
@@ -0,0 +1,130 @@
+The moz-fx-data-shared-prod project is assumed because shredder currently only runs
+on tables in this project.
+
+Use the dry run parameter run shredder with the --dry-run option to validate parameters.
+Note that the shredder dry run will still dry run queries against every partition of each table
+so it may take a long time to finish if a lot of tables are given.
+
+This DAG is meant to be used to handle older deletion requests for tables that are already being
+shredded.  Any provided tables that aren't already valid deletion targets will be ignored.
+
+#### Owner
+
+bewu@mozilla.com
+"""
+
+params = {
+    "request_start_date": Param(
+        default=(date.today() - timedelta(days=7)).isoformat(),
+        description="First date of deletion requests to process",
+        type="string",
+        format="date",
+    ),
+    "request_end_date": Param(
+        default=(date.today()).isoformat(),
+        description="Last date of data (i.e. partition) to delete from",
+        type="string",
+        format="date",
+    ),
+    "target_tables": Param(
+        default=["dataset.table_name"],
+        description="Tables to delete from (one per line)",
+        type="array",
+        minItems=1,
+    ),
+    "dry_run": Param(default=True, type="boolean"),
+}
+
+default_args = {
+    "owner": "bewu@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2024, 3, 1),
+    "catchup": False,
+    "email": [
+        "telemetry-alerts@mozilla.com",
+        "bewu@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": False,
+    # transient failures are expected and can be handled with state table
+    "retries": 44,
+    "retry_delay": timedelta(minutes=5),
+}
+
+tags = [
+    Tag.ImpactTier.tier_3,
+    Tag.Triage.no_triage,
+]
+
+NON_DRY_RUN_TASK_ID = "shredder_backfill"
+DRY_RUN_TASK_ID = "shredder_backfill_dry_run"
+
+
+def base_backfill_operator(dry_run):
+    """Create task for backfill, filling out parameters based on dry run."""
+    return GKEPodOperator(
+        task_id=DRY_RUN_TASK_ID if dry_run else NON_DRY_RUN_TASK_ID,
+        cmds=[
+            "script/shredder_delete",
+            *(["--dry-run"] if dry_run else []),
+            # use different tables from scheduled task so they can be monitored separately
+            "--state-table=moz-fx-data-shredder.shredder_state.shredder_state_backfill",
+            "--task-table=moz-fx-data-shredder.shredder_state.tasks_backfill",
+            "--end-date={{ params.request_end_date }}",
+            "--start-date={{ params.request_start_date }}",
+            "--no-use-dml",
+            # low parallelism to reduce slot contention with scheduled task
+            "--parallelism=1",
+            "--billing-project=moz-fx-data-bq-batch-prod",
+            "--only",
+        ],
+        # target_tables will be rendered as a python list
+        arguments="{{ params.target_tables }}",
```

Comment:
I checked that this doesn't allow for command injection by passing in something like `; echo 123`, although this would work for the `cmds` parameter.